Koans/AboutReflection.cs

```diff
@@ -0,0 +1,122 @@
+using Xunit;
+using System;
+using System.Globalization;
+using DotNetKoans.Engine;
+using System.Reflection;
+using System.Linq;
+using System.Diagnostics;
+using System.Reflection.Emit;
+
+namespace DotNetKoans.Koans;
+
+public class AboutReflection : Koan
+{
+	// Reflection enables you to obtain information about loaded assemblies and types defined within them
+	// You can use reflection to create type instances at run time, and to invoke and access them
+	// Assemblies contain modules, modules contin types and types contain members.
+	// Reflection provides objects that encapsulate assemblies, modules and types.
+	// You can use reflection to dynamically create an instance of a type, bind the type to an existing object,
+	// or get the type from an existing object.
+	// You can then invoke the type's methods or access its fields and properties.
+
+	[Step(1)]
+	public void GettingAType()
+	{
+		// Everything in .net is derived from object
+		// object contains the method GetType(), which will return its type
+		// this is the most basic form of reflection
+
+		object thisIsAStringType = "a string";
+		Assert.Equal(typeof(FillMeIn), thisIsAStringType.GetType());
+	}
+
+	[Step(2)]
+	public void ViewingTheAssemblyOfAnObject()
+	{
+		// All .net projects (.csproj), when compiled are called assemblies
+		// The GetName() method will return the name of the assembly that contains this class, which will be the same as the .csproj file
+		
+		Assembly a = typeof(AboutReflection).Module.Assembly;
+
+		Assert.Equal(FILL_ME_IN, a.GetName().Name);
+	}
+
+	[Step(3)]
+	public void InspectingTheTypesInAnExternalAssembly()
+	{
+		// By now you will have noticed that DotNetKoans is cleverly using Xunit assertions as its teaching mechanic
+		// Xunit is an example of an external assembly installed via package manager
+		// Using reflection we can inspect its types
+		// Hint: we're using a type that we already know about to get the Xunit assembly and all of its types
+
+		Assembly a = typeof(Assert).Module.Assembly;
+
+		Type[] types = a.GetTypes();
+
+		Assert.Contains(typeof(FillMeIn), types);
```

Comment:
What should the user enter here?

Reply:
It would have been `Assert`. I was trying to show off peeking into external assemblies/packages here, but it probably isn't clear enough.

Reply:
Mmh, maybe you can make it more clear.
On the other hand, I must say that it was late yesterday and I totally did not see the `Contains` and read `Equals` instead. I then accidentally deleted the line in my commit 🤦🏽‍♂️